lib/cocoapods/installer/xcode/target_validator.rb

```diff
@@ -71,16 +71,22 @@ def verify_no_static_framework_transitive_dependencies
             next unless aggregate_target.requires_frameworks?
 
             aggregate_target.user_build_configurations.keys.each do |config|
-              pod_targets = aggregate_target.pod_targets_for_build_configuration(config)
+              dynamic_pod_targets = aggregate_target.pod_targets_for_build_configuration(config).reject(&:static_framework?)
 
-              dependencies = pod_targets.select(&:should_build?).reject(&:static_framework?).flat_map(&:dependencies)
-              depended_upon_targets = pod_targets.select { |t| dependencies.include?(t.pod_name) && !t.should_build? }
+              dependencies = dynamic_pod_targets.select(&:should_build?).flat_map(&:dependencies)
+              depended_upon_targets = dynamic_pod_targets.select { |t| dependencies.include?(t.pod_name) && !t.should_build? }
 
               static_libs = depended_upon_targets.flat_map(&:file_accessors).flat_map(&:vendored_static_artifacts)
               unless static_libs.empty?
                 raise Informative, "The '#{aggregate_target.label}' target has " \
                   "transitive dependencies that include static binaries: (#{static_libs.to_sentence})"
               end
+
+              static_framework_deps = dynamic_pod_targets.select(&:should_build?).flat_map(&:recursive_dependent_targets).select(&:static_framework?)
```

Comment:
in theory, it's ok for this to not be empty, so long as each static framework is only linked into _exactly one_ dynamic binary -- that binary need not be the user target

Reply:
Maybe in theory - although I'm not a 100% convinced.  Is a compile for a static object always relocatable? It's not necessarily so and definitely isn't in some ABIs. Even so, it's not supported by the implementation as demonstrated by #7352.

I propose we go ahead with this PR since it makes static_framework transitive checking consistent with binary vendored static frameworks.

We could follow up on this point in a future PR.

Reply:
Yeah, this is good as is! Just wanted to point out that edge case